src/pybind/mgr/dashboard/services/rbd.py

```diff
@@ -330,14 +330,28 @@ def _rbd_image(cls, ioctx, pool_name, namespace, image_name):
             return stat
 
     @classmethod
-    def _rbd_image_names(cls, ioctx):
+    def _rbd_image_refs(cls, ioctx):
         rbd_inst = rbd.RBD()
-        return rbd_inst.list(ioctx)
+        return rbd_inst.list2(ioctx)
 
     @classmethod
     def _rbd_image_stat(cls, ioctx, pool_name, namespace, image_name):
         return cls._rbd_image(ioctx, pool_name, namespace, image_name)
 
+    @classmethod
+    def _rbd_image_stat_removing(cls, ioctx, pool_name, namespace, image_id):
+        rbd_inst = rbd.RBD()
+        img = rbd_inst.trash_get(ioctx, image_id)
+
+        if img['source'] == 'REMOVING':
+            img['unique_id'] = get_image_spec(pool_name, namespace, image_id)
+            img['pool_name'] = pool_name
+            img['namespace'] = namespace
+            img['deletion_time'] = "{}Z".format(img['deletion_time'].isoformat())
+            img['deferment_end_time'] = "{}Z".format(img['deferment_end_time'].isoformat())
+            return img
+        return None
```

Comment:
Just a nit so feel free to ignore but since you are expecting `rbd.ImageNotFound` from this function anyway, perhaps throw that exception instead of returning `None`?  Then both `ENOENT` and `!REMOVING` cases would be handled by the same `continue` in `rbd_pool_list()`.

Reply:
Hmmm... good idea... but I think `ImageNotFound` is not the appropriate exception, is it?

Reply:
The function name suggests that it is looking for an image in `REMOVING` state and for the case when it is not finding it `ImageNotFound` sounds appropriate to me.

Reply:
Yes, I agree. From the perspective it's looking for an RBD with `id == xy` AND `source == REMOVING` it makes sense to me. I just thought about the RBD exists and might not be returned by the method though (`source != REMOVING`).